packages/protocol/test/governance/election.ts

```diff
@@ -0,0 +1,854 @@
+      beforeEach(async () => {})
+
+      it("should return that group's member list", async () => {
+        await election.vote(group1, voter1.weight, group2, NULL_ADDRESS, { from: voter1.address })
+        assertSameAddresses(await election.electValidators(), [
+          validator1,
+          validator2,
+          validator3,
+          validator4,
+        ])
+      })
+    })
+
+    describe("when > maxElectableValidators members's groups receive votes", () => {
+      beforeEach(async () => {
+        await election.vote(group1, voter1.weight, group2, NULL_ADDRESS, { from: voter1.address })
+        await election.vote(group2, voter2.weight, NULL_ADDRESS, group1, { from: voter2.address })
+        await election.vote(group3, voter3.weight, NULL_ADDRESS, group2, { from: voter3.address })
+      })
+
+      it('should return maxElectableValidators elected validators', async () => {
+        assertSameAddresses(await election.electValidators(), [
+          validator1,
+          validator2,
+          validator3,
+          validator5,
+          validator6,
+          validator7,
+        ])
+      })
+    })
+
+    describe('when different random values are provided', () => {
+      beforeEach(async () => {
+        await election.vote(group1, voter1.weight, group2, NULL_ADDRESS, { from: voter1.address })
+        await election.vote(group2, voter2.weight, NULL_ADDRESS, group1, { from: voter2.address })
+        await election.vote(group3, voter3.weight, NULL_ADDRESS, group2, { from: voter3.address })
+      })
+
+      it('should return different results', async () => {
+        await random.setRandom(hash1)
+        const valsWithHash1 = (await election.electValidators()).map((x) => x.toLowerCase())
+        await random.setRandom(hash2)
+        const valsWithHash2 = (await election.electValidators()).map((x) => x.toLowerCase())
+        assert.sameMembers(valsWithHash1, valsWithHash2)
+        assert.notDeepEqual(valsWithHash1, valsWithHash2)
+      })
+    })
+
+    describe('when a group receives enough votes for > n seats but only has n members', () => {
+      beforeEach(async () => {
+        // By incrementing the total votes by 80, we allow group3 to receive 80 votes from voter3.
+        const increment = 80
+        const votes = 80
+        await mockLockedGold.incrementNonvotingAccountBalance(voter3.address, increment)
+        await mockLockedGold.setTotalLockedGold(totalLockedGold + increment)
+        await election.vote(group3, votes, group2, NULL_ADDRESS, { from: voter3.address })
+        await election.vote(group1, voter1.weight, NULL_ADDRESS, group3, { from: voter1.address })
+        await election.vote(group2, voter2.weight, NULL_ADDRESS, group1, { from: voter2.address })
+      })
+
+      it('should elect only n members from that group', async () => {
+        assertSameAddresses(await election.electValidators(), [
+          validator7,
```

Comment:
Did you mean to have these out of order?

Reply:
I believe they're ordered in the order that they would be elected by d'hondt